pkg/scheduler/core/generic_scheduler.go

```diff
@@ -525,7 +515,15 @@ func (g *genericScheduler) findNodesThatFit(pluginContext *framework.PluginConte
 				}
 			} else {
 				predicateResultLock.Lock()
-				failedPredicateMap[nodeName] = failedPredicates
+				if status != nil && !status.IsSuccess() {
```

Comment:
nit: `!status.IsSuccess()` won't panic even if the status is nil, please take a look at the implementation of `IsSuccess` function.

Reply:
Sorry, I understand that you don't mean it. You mean that when the status is nil, calling `status.IsSusccess()` won't panic? 
`nil` is untyped, my understanding will get an error similar to `use of untyped nil` panic error.

If my understanding is wrong, please let me know. Thanks. :)

Reply:
> Sorry, I understand that you don't mean it. You mean that when the status is nil, calling `status.IsSusccess()` won't panic?

Please take a look at https://github.com/kubernetes/kubernetes/pull/81876/files#diff-c237cdd9e4cb201118ca380732d7f361L509

Reply:
I created a [snippet](https://play.golang.org/p/GYoTeiT5hFi) here which shows it won't cause panic.

```go
package main

import (
	"fmt"
)

type Status struct{}

func (s *Status) IsSuccess() bool {
	if s == nil {
		return true
	}
	return false
}

func main() {
	st := returnStatus()
	fmt.Println(st.IsSuccess())
}

func returnStatus() *Status {
	return nil
}
package main

import (
	"fmt"
)

type Status struct{}

func (s *Status) IsSuccess() bool {
	if s == nil {
		return true
	}
	return false
}

func main() {
	st := returnStatus()
	fmt.Println(st.IsSuccess())
}

func returnStatus() *Status {
	return nil
}

$ go run ...
true
```

Reply:
cool! I got it.